### PR TITLE
dvc.lock updates

### PR DESCRIPTION
**Pull request by jorgeorpinel**
Continuing #2131

### CODE REVIEW

**Review by shcheklein — APPROVED** [review]

File: content/docs/user-guide/project-structure/pipelines-files.md [hunk 1]
```diff
@@ -364,7 +364,11 @@ validation and auto-completion.
 > See also
 > [How to Merge Conflicts](/doc/user-guide/how-to/merge-conflicts#dvcyaml).
 
-### Output entries
+### Simple output entries
```
Comment (shcheklein) [inline content/docs/user-guide/project-structure/pipelines-files.md:367]: hmm, I don't like this term "simple". There is nothing specifically simple or complex about those entries. It's totally fine to have them defined differently, with a bit different schema if we very clear in docs about this.

Comment (jorgeorpinel) [reply content/docs/user-guide/project-structure/pipelines-files.md:367]: Agree. But the reason I changed the title is that the output (and dependency) entries of dvc.lock files are = those in .dvc files, not these (dvc.yaml output entries). And in the dvc.lock section below I use the terms "out/dep entry", linked to the corresponding section in the .dvc file guide, to avoid repeating those sections in this doc again.

Repeating here is a good option and probably even less confusing in some ways, but we would still need to somehow emphasize that dvc.yaml outs/deps entries are different from those in dvc.lock. Probably we would do that with different section names anyway.

TLDR: How about *dvc.yaml output entries* and **dvc.lock output entries**?

Comment (shcheklein) [reply content/docs/user-guide/project-structure/pipelines-files.md:367]: To my mind we are already in the section that describes the dvc.yaml schema and ideally it should be clear from the context that this is about `dvc.yaml`. I can see how additional message can help, but anything that comes to my mind would create even more confusion (people are not aware about other files, different types of outputs, etc).

Comment (jorgeorpinel) [reply content/docs/user-guide/project-structure/pipelines-files.md:367]: You know what I looked into this and I think I have an easier solution ⌛

Comment (jorgeorpinel) [reply content/docs/user-guide/project-structure/pipelines-files.md]: OK I simplified by calling the section in dvc.yaml "Output subfields" (which is what it is), PTAL.

If it's still confusing we could consider calling the sections in .dvc "Output/Dependency **tracking** entries".

Comment (jorgeorpinel) [reply content/docs/user-guide/project-structure/pipelines-files.md]: p.s. TBH the ideal solution would be to have tables inside tables but that's not doable on md, would need some design and frontend work.

Comment (shcheklein) [reply content/docs/user-guide/project-structure/pipelines-files.md]: sounds good!

**Comment by jorgeorpinel** [comment]
OK, merging then 🙂
